### PR TITLE
Switch to Grid and Footer

### LINKED ISSUE #83: Update Footer
Should match frequency.xyz

image

### PR DESCRIPTION
**Pull request by wilwade**
Problem
=======

Needed to do the footer, but had to change a good amount to make it work.

Closes #83 

Solution
========

- Added a grid
- Moved the old menu icons to the left as a column
- Shifted around a few things for the grid.
- Kept the resize
- Kept the sidebar animation
- Kept the touch controls for mobile (swipe to get the sidebar)
- Added the footer

Screenshots (optional):
-----------------------
Here are a few, but you really have pull and run locally

Screen Shot 2024-11-04 at 11 59 29-fullpage
Screen Shot 2024-11-04 at 11 59 47-fullpage
Screen Shot 2024-11-04 at 12 00 07-fullpage
Screen Shot 2024-11-04 at 12 00 17-fullpage
image

image

image

### CODE REVIEW

File: css/header.css [hunk 1]
```diff
@@ -9,27 +9,30 @@
...
+  background-color: var(--header-background);
+  box-shadow: 0 4px 4px 0 #00000040;
+  width: 100vw;
+  z-index: 1000;
 }
 
 @media (min-width: 800px) {
   .frqcy-header {
-    padding: 0 80px;
+    padding: 0 62px;
```
Comment (wilwade) [inline css/header.css:35]: This helps align the Frequency logo by default with the sidebar

File: theme/book.js [hunk 2]
```diff
@@ -604,66 +604,3 @@ function playground_text(playground, hidden = true) {
     document.scrollingElement.scrollTo({ top: 0, behavior: "smooth" });
   });
 })();
-
```
Comment (wilwade) [inline theme/book.js:607]: We don't want to do any of this anymore. Perhaps we'll bring back the scrolling header disappearing later.

File: theme/css/chrome.css [hunk 3]
```diff
@@ -2,6 +2,53 @@
 
 @import "variables.css";
 
+:target {
+  scroll-margin-top: calc(var(--header-height) + 20px);
```
Comment (wilwade) [inline theme/css/chrome.css:6]: Make it so that the anchor links aren't hidden by the header.

File: theme/css/chrome.css [hunk 4]
```diff
@@ -2,6 +2,53 @@
...
+}
+
+/* GRID */
```
Comment (wilwade) [inline theme/css/chrome.css:9]: Had to move to grid to make everything work. No, I didn't want to, but it was the cleanest way.

File: theme/css/chrome.css [hunk 5]
```diff
@@ -34,44 +81,22 @@ a > .hljs {
         will want to reposition the viewport in a weird way.
     */
   overflow-x: clip;
+  transition: 200ms;
 }
 
 /* Menu Bar */
```
Comment (wilwade) [inline theme/css/chrome.css:87]: Menu bar changes removed a ton of css.

File: theme/css/chrome.css [hunk 6]
```diff
@@ -454,22 +455,24 @@ ul#searchresults span.teaser em {
...
   font-size: 0.875em;
   box-sizing: border-box;
-  -webkit-overflow-scrolling: touch;
-  overscroll-behavior-y: contain;
   background-color: var(--sidebar-bg);
   color: var(--sidebar-fg);
+  position: relative;
 }
 
-[dir="rtl"] .sidebar {
```
Comment (wilwade) [inline theme/css/chrome.css:470]: Not currently looking to move the sidebar side for rtl. Would require grid setup

File: theme/css/chrome.css [hunk 7]
```diff
@@ -682,11 +650,11 @@ ul#searchresults span.teaser em {
 .theme-popup {
   position: absolute;
   left: 10px;
-  top: var(--menu-bar-height);
+  top: 0;
```
Comment (wilwade) [inline theme/css/chrome.css:653]: This way the theme popup is with the icon

File: theme/index.hbs [hunk 8]
```diff
@@ -154,82 +158,104 @@
             }
         </script>
 
-        {{> header}}
-
-        <div id="page-wrapper" class="page-wrapper">
-
-            <div class="page">
```
Comment (wilwade) [inline theme/index.hbs:161]: Removed this extra div, but otherwise most of these are moving things around.

File: theme/index.hbs [hunk 9]
```diff
@@ -154,82 +158,104 @@
...
-                <div id="menu-bar-hover-placeholder"></div>
-                <div id="menu-bar" class="menu-bar sticky">
```
Comment (wilwade) [inline theme/index.hbs:163]: Moved outside to the grid structure

**Review by saraswatpuneet — APPROVED** [review]
Looks good to me, ran locally

The inner container seems to be sizing in and out based on content. readability is not consistent, suggestion would be to pin the forward and back button pin to bottom of page, or pin footer

**Review by shannonwells — APPROVED** [review]
All works for me, but I have a couple of notes, which I don't consider blocking until we get some more finalized design:
* Was there supposed to be a dark version of the footer?

* I don't know if this is a regression, but the prev/next chapter buttons are off-center for this view:
Screenshot 2024-11-04 at 1 32 23 PM

**Pull request merged**